airflow/providers/zendesk/hooks/zendesk.py

```diff
@@ -31,90 +32,91 @@ class ZendeskHook(BaseHook):
     :param zendesk_conn_id: The Airflow connection used for Zendesk credentials.
     """
 
-    def __init__(self, zendesk_conn_id: str) -> None:
+    conn_name_attr = 'zendesk_conn_id'
+    default_conn_name = 'zendesk_default'
+    conn_type = 'zendesk'
+    hook_name = 'Zendesk'
+
+    def __init__(self, zendesk_conn_id: str = default_conn_name) -> None:
         super().__init__()
-        self.__zendesk_conn_id = zendesk_conn_id
-        self.__url = None
+        self.zendesk_conn_id = zendesk_conn_id
+        self.base_api: Optional[BaseApi] = None
+        zenpy_client, url = self._init_conn()
+        self.zenpy_client = zenpy_client
+        self.__url = url
+        # Keep reference to the _get method to allow arbitrary endpoint call for
+        # backwards compatibility. (alternative to old ZendeskHook.call method)
+        self.get = self.zenpy_client.users._get
```

Comment:
users may not be familiar with the old lib.
Here we should just explain what this parameter is for (or preferably in example dag)
The explanation about backward compatibility is already written in the change log

Reply:
Yes no worries. I have removed this comment and added a simple example DAG to illustrate this. (also added the `example DAGs` link to the doc)